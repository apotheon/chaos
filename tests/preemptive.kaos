void def f51()
    print b
end

function_table

num def f1()
    num a = 101
    return a
end

num def f2()
    num b = 102
    return b
end

num def f3()
    num c = 103
    return c
end

num def f52(num x, num y)
    num z = x + y
    bool a = true
end {
    z == 8  : f1(),
    n1 > 10 : f2(),
    default : f3()
}

function_table


num def f53(num x, num y)
    num z = x + y
    bool a = true
end {
    z == 8  : f1(),
    z > 10  : f2(n2),
    default : f3()
}

function_table


void def a()
    num b = f5()
end

function_table


num def f54(num x, num y)
    num z = x + y
    bool a = true
end {
    z == 8  : f1(),
    z > 10  : f6(),
    default : f3()
}

function_table


num def f55(num x, num y)
    num z = x + y
    bool a = true
end {
    z == 8  : f1(z),
    z > 10  : f2(),
    default : f3()
}

function_table


num def f56()
    num c = 103
    num x = f1(c)
    return c
end

function_table


num def f8(num a, str b)
    print a
    print b
    num c = a + 5
    return c
end

function_table

void def f57()
    num a = 3
    num b = 5
    f8(a, b)
end

function_table

void def f58()
    f8(3, 5)
end

function_table

void def f59()
    list a = [1, 2, 3]
    str b = 'hello'
    f8(a, b)
end

function_table

void def f60()
    f8([1, 2, 3], 'hello')
end

function_table

void def f61()
    num a = 3
    str b = 'hello'
    bool c = true
    f8(a, b, c)
end

function_table

void def f62()
    f8(3, 'hello', true)
end

function_table


list def f63()
    dict a = {'a': 1, 'b': 2, 'c': 3}
    return a
end

function_table

dict def f64()
    dict a = {'a': 1, 'b': 2, 'c': 3}
    return b
end

function_table


list def f65()
    num x = 5
    dict a = {'a': 1, 'b': 2, 'c': 3}
end {
    x == 5  : return a,
    default : return a
}

function_table

dict def f66()
    num x = 5
    dict a = {'a': 1, 'b': 2, 'c': 3}
end {
    x == 5  : return b,
    default : return b
}

function_table

dict def f67()
    num x = 5
end

function_table


void def f68()
    3 times do
        print b
        2 times do
            print c
        end
    end
end

function_table

void def f69()
    n times do
        print "hello world"
    end
end

function_table

void def f70()
    foreach a as el
        print el
    end
end

function_table

void def f7()
    list a = [1, 2, 3]
    foreach a as el
        print el
    end
end

function_table

void def f71()
    list a = [1, 2, 3]
    foreach a as el
        print el
        print b
    end
end

function_table

void def f8()
    dict a = {'a': 1, 'b': 2, 'c': 3}
    foreach a as key : val
        print key
        print val
    end
end

function_table

void def f72()
    dict a = {'a': 1, 'b': 2, 'c': 3}
    foreach a as key : val
        print key
        print val
        print b
    end
end

function_table


num def f9(num x, num y)
    num z = x + y
    bool a = true
end {
    z > 5   : break,
    z == 8  : f1(),
    z > 10  : f2(),
    default : f3()
}

function_table

void def f73(num list a)
    foreach a as el
        print el
        print f11(el, 1)
    end
end

function_table

void def f74(num a)
    foreach a as el
        print el
        print f9(el, 1)
    end
end

function_table

void def f75(num a)
    foreach a as key : val
        print key
        print f9(val, 1)
    end
end

function_table


num def f76(num x, num y)
    num r = f9(x, y)
    return r
end

function_table

void def f12(num list a)
    foreach a as el
        print el
        print f9(el, 1)
    end
end

function_table


void def f13(num list a)
    foreach a as el
        print f9(el, 1)
    end
end

list a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]

f13(a)
