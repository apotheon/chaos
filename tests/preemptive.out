[1;43m  Preemptive Error:                                   [0m
[0;43m    File: "tests/preemptive.kaos", line 2, in f51     [0m
[0;43m      print b                                         [0m
[1;43m  Undefined variable: b                               [0m
  ---
[1;43m  Preemptive Error:                                    [0m
[0;43m    File: "tests/preemptive.kaos", line 27, in f52     [0m
[0;43m      n1 > 10 : f2(),                                  [0m
[1;43m  Undefined variable: n1                               [0m
  ---
[1;43m  Preemptive Error:                                    [0m
[0;43m    File: "tests/preemptive.kaos", line 39, in f53     [0m
[0;43m      z > 10  : f2(n2),                                [0m
[1;43m  Undefined variable: n2                               [0m
  ---
[1;43m  Preemptive Error:                                  [0m
[0;43m    File: "tests/preemptive.kaos", line 47, in a     [0m
[0;43m      num b = f5()                                   [0m
[1;43m  Undefined function: f5 in <module>                 [0m
  ---
[1;43m  Preemptive Error:                                    [0m
[0;43m    File: "tests/preemptive.kaos", line 58, in f54     [0m
[0;43m      z > 10  : f6(),                                  [0m
[1;43m  Undefined function: f6 in <module>                   [0m
  ---
[1;43m  Preemptive Error:                                   [0m
[0;43m    File: "tests/preemptive.kaos", line 69, in f1     [0m
[0;43m      z == 8  : f1(z),                                [0m
[1;43m  Incorrect argument count for function: f1           [0m
  ---
[1;43m  Preemptive Error:                                   [0m
[0;43m    File: "tests/preemptive.kaos", line 79, in f1     [0m
[0;43m      num x = f1(c)                                   [0m
[1;43m  Incorrect argument count for function: f1           [0m
  ---
[1;43m  Preemptive Error:                                   [0m
[0;43m    File: "tests/preemptive.kaos", line 98, in f8     [0m
[0;43m      f8(a, b)                                        [0m
[1;43m  Incorrect argument count for function: f8           [0m
  ---
[1;43m  Preemptive Error:                                    [0m
[0;43m    File: "tests/preemptive.kaos", line 104, in f8     [0m
[0;43m      f8(3, 5)                                         [0m
[1;43m  Incorrect argument count for function: f8            [0m
  ---
[1;43m  Preemptive Error:                                    [0m
[0;43m    File: "tests/preemptive.kaos", line 112, in f8     [0m
[0;43m      f8(a, b)                                         [0m
[1;43m  Incorrect argument count for function: f8            [0m
  ---
[1;43m  Preemptive Error:                                    [0m
[0;43m    File: "tests/preemptive.kaos", line 118, in f8     [0m
[0;43m      f8([1, 2, 3], 'hello')                           [0m
[1;43m  Incorrect argument count for function: f8            [0m
  ---
[1;43m  Preemptive Error:                                    [0m
[0;43m    File: "tests/preemptive.kaos", line 127, in f8     [0m
[0;43m      f8(a, b, c)                                      [0m
[1;43m  Incorrect argument count for function: f8            [0m
  ---
[1;43m  Preemptive Error:                                    [0m
[0;43m    File: "tests/preemptive.kaos", line 133, in f8     [0m
[0;43m      f8(3, 'hello', true)                             [0m
[1;43m  Incorrect argument count for function: f8            [0m
  ---
[1;43m  Preemptive Error:                                       [0m
[0;43m    File: "tests/preemptive.kaos", line 141, in f63       [0m
[0;43m      return a                                            [0m
[1;43m  Illegal variable type: Dictionary for function: f63     [0m
  ---
[1;43m  Preemptive Error:                                     [0m
[0;43m    File: "tests/preemptive.kaos", line 148, in f64     [0m
[0;43m      return b                                          [0m
[1;43m  Undefined variable: b                                 [0m
  ---
[1;43m  Preemptive Error:                                       [0m
[0;43m    File: "tests/preemptive.kaos", line 158, in f65       [0m
[0;43m      x == 5  : return a,                                 [0m
[1;43m  Illegal variable type: Dictionary for function: f65     [0m
  ---
[1;43m  Preemptive Error:                                     [0m
[0;43m    File: "tests/preemptive.kaos", line 168, in f66     [0m
[0;43m      x == 5  : return b,                               [0m
[1;43m  Undefined variable: b                                 [0m
  ---
[1;43m  Preemptive Error:                                     [0m
[0;43m    File: "tests/preemptive.kaos", line 176, in f67     [0m
[0;43m      end                                               [0m
[1;43m  The function 'f67' did not return anything!           [0m
  ---
[1;43m  Preemptive Error:                                     [0m
[0;43m    File: "tests/preemptive.kaos", line 183, in f68     [0m
[0;43m      print b                                           [0m
[1;43m  Undefined variable: b                                 [0m
  ---
[1;43m  Preemptive Error:                                     [0m
[0;43m    File: "tests/preemptive.kaos", line 193, in f69     [0m
[0;43m      n times do                                        [0m
[1;43m  Undefined variable: n                                 [0m
  ---
[1;43m  Preemptive Error:                                     [0m
[0;43m    File: "tests/preemptive.kaos", line 201, in f70     [0m
[0;43m      foreach a as el                                   [0m
[1;43m  Undefined variable: a                                 [0m
  ---
[1;43m  Preemptive Error:                                     [0m
[0;43m    File: "tests/preemptive.kaos", line 221, in f71     [0m
[0;43m      print b                                           [0m
[1;43m  Undefined variable: b                                 [0m
  ---
[1;43m  Preemptive Error:                                     [0m
[0;43m    File: "tests/preemptive.kaos", line 242, in f72     [0m
[0;43m      print b                                           [0m
[1;43m  Undefined variable: b                                 [0m
  ---
[1;43m  Preemptive Error:                                     [0m
[0;43m    File: "tests/preemptive.kaos", line 264, in f73     [0m
[0;43m      print f11(el, 1)                                  [0m
[1;43m  Undefined function: f11 in <module>                   [0m
  ---
[1;43m  Preemptive Error:                                     [0m
[0;43m    File: "tests/preemptive.kaos", line 271, in f74     [0m
[0;43m      foreach a as el                                   [0m
[1;43m  'a' is not a list!                                    [0m
  ---
[1;43m  Preemptive Error:                                     [0m
[0;43m    File: "tests/preemptive.kaos", line 280, in f75     [0m
[0;43m      foreach a as key : val                            [0m
[1;43m  'a' is not a dictionary!                              [0m
  ---
[1;43m  Preemptive Error:                                             [0m
[0;43m    File: "tests/preemptive.kaos", line 290, in f9              [0m
[0;43m      num r = f9(x, y)                                          [0m
[1;43m  Call to a function with `break` from a multiline loop: f9     [0m
  ---
[1;43m  Preemptive Error:                                             [0m
[0;43m    File: "tests/preemptive.kaos", line 299, in f9              [0m
[0;43m      print f9(el, 1)                                           [0m
[1;43m  Call to a function with `break` from a multiline loop: f9     [0m
